sky/provision/kubernetes/network_utils.py

```diff
@@ -19,9 +20,16 @@
 def get_port_mode(
         mode_str: Optional[str] = None) -> kubernetes_enums.KubernetesPortMode:
     """Get the port mode from the provider config."""
-    mode_str = mode_str or skypilot_config.get_nested(
-        ('kubernetes', 'ports'),
-        kubernetes_enums.KubernetesPortMode.LOADBALANCER.value)
+    curr_context = kubernetes_utils.get_current_kube_config_context_name()
+    running_kind =  curr_context == kubernetes_utils.KIND_CONTEXT_NAME
+    mode_str = (
+        (running_kind and kubernetes_enums.KubernetesPortMode.INGRESS.value) or
+        mode_str or
+        skypilot_config.get_nested(
+            ('kubernetes', 'ports'),
+            kubernetes_enums.KubernetesPortMode.LOADBALANCER.value
+        )
+    )
```

Comment:
As discussed on slack, good to remove this and stick to config.yaml as the ground truth for port mode.